api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirEncounterDaoImpl.java

```diff
@@ -9,41 +9,31 @@
  */
 package org.openmrs.module.fhir2.api.dao.impl;
 
-import static org.hibernate.criterion.Restrictions.eq;
-
 import java.util.Collection;
 
 import ca.uhn.fhir.rest.param.DateRangeParam;
 import ca.uhn.fhir.rest.param.ReferenceAndListParam;
 import lombok.AccessLevel;
 import lombok.Setter;
 import org.hibernate.Criteria;
-import org.hibernate.SessionFactory;
 import org.openmrs.Encounter;
 import org.openmrs.module.fhir2.api.dao.FhirEncounterDao;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Component;
 
 @Component
 @Setter(AccessLevel.PACKAGE)
-public class FhirEncounterDaoImpl extends BaseDaoImpl implements FhirEncounterDao {
-	
-	@Autowired
-	@Qualifier("sessionFactory")
-	private SessionFactory sessionFactory;
+public class FhirEncounterDaoImpl extends BaseFhirDaoImpl<Encounter> implements FhirEncounterDao {
 	
 	@Override
-	public Encounter getEncounterByUuid(String uuid) {
-		return (Encounter) sessionFactory.getCurrentSession().createCriteria(Encounter.class).add(eq("uuid", uuid))
-		        .uniqueResult();
+	public Encounter get(String uuid) {
+		return super.get(uuid);
 	}
```

Comment:
You shouldn't need to implement this method at all.

Reply:
@ibacher  thanks for the review ,,
i getting abit confused here why the method should be implemented since its coming from the super class being extended and its only refactoring an already existing method with a change in the method name ..
Just requesting for some more explanation here to help me better understand ..Thanks

Reply:
It doesn't need to be here at all. I think it's left in the `FhirEncounterDao` interface so we can include Javadoc documenting its use, but in the actual implementation, we can rely on the super class version.

Reply:
thanks ,,have fixed it ..